src/day2/scan_passwds.rb

```diff
@@ -0,0 +1,95 @@
+#! /usr/bin/env -S ruby -w
+
+#: AoC Day 2 - Count valid passwords. (See `problems.txt` for the exact
+#: problems definitions.)
+
+require 'optparse'
+require 'ostruct'
+
+module ScanPasswds
+
+  #: Return true if given passwd matches original shopkeeper's policies
+  #: constraints, false otherwise.
+  def ScanPasswds.match_constraints_orig?(min, max, char, passwd)
+    count = passwd.count char
+    min <= count && count <= max
+  end
+
+  #: Return true if given passwd matches Toboggan Corporate Policies constraints,
+  #: false otherwise.
+  def ScanPasswds.match_constraints_tcp?(fst_idx, sec_idx, char, passwd)
```

Comment:
Don't abbreviate variables. We consider this a bad practice.

Reply:
Assuming you mean the arguments to `match_constraints_tcp?` - fixed in df1b427.